### PR TITLE
update rekor intoto/0.0.2 entry hash calculation

### PR DESCRIPTION
**Pull request by bdehamer**
Signed-off-by: Brian DeHamer <bdehamer@github.com>

#### Summary

Updates the logic to properly calculate the digest of a DSSE envelope when creating an intoto/0.0.2 style entry in Rekor.

### CODE REVIEW

File: src/tlog/format.ts [hunk 1]
```diff
@@ -111,6 +117,28 @@ function toProposedIntotoV002Entry(
   };
 }
 
+// Calculates the hash of a DSSE envelope for inclusion in a Rekor entry.
+// There is no standard way to do this, so the scheme we're using as as
+// follows:
+//  * payload is base64 encoded
+//  * signature is base64 encoded (only the first signature is used)
+//  * keyid is included ONLY if it is NOT an empty string
+//  * The resulting JSON is canonicalized and hashed to a hex string
```
Comment (feelepxyz) [inline src/tlog/format.ts:126]: Yikes!

Comment (feelepxyz) [reply src/tlog/format.ts:126]: Is it worth writing some kind of tests around this behaviour to make sure we don't deviate from it in future?

Comment (feelepxyz) [reply src/tlog/format.ts:126]: Do we have some tests in src/tlog/verify.test.ts that create different kinds of envelopes with and without a keyid? Maybe with multiple sigs?

**Review by feelepxyz — COMMENTED** [review]
Gnarly.. nice work figuring this out! Just had one comment around tests, otherwise LGTM 👍

**Review by codysoyland — DISMISSED** [review]
🎉

**Review by feelepxyz — APPROVED** [review]
Thanks for adding the tests!

**Pull request merged**